torchtune/modules/attention.py

```diff
@@ -117,6 +117,37 @@ def __init__(
         self.output_proj = output_proj
         self.pos_embeddings = pos_embeddings
 
+    def setup_cache(self, batch_size: int, dtype: torch.dtype) -> None:
+        """Setup key value caches for attention calculation.
+
+        Args:
+            batch_size (int): batch size for the caches.
+            dtype (torch.dtype): dtype for the caches.
+
+        Raises:
+            RuntimeError: if called when kv_cache is already setup.
+        """
+        # Don't overwrite user defined kv_cache from init
+        if self.kv_cache is not None:
+            raise RuntimeError(
```

Comment:
should this throw an error or just raise a warning and skip setup if called twice?